app/routes/circulars/circulars.lib.ts

```diff
@@ -143,3 +161,12 @@ export function formatAuthor({
   else if (!affiliation) return `${name} <${email}>`
   else return `${name} at ${affiliation} <${email}>`
 }
+
+export function parseEventFromSubject(value: string) {
+  for (const [regexp, normalize] of subjectMatchers) {
+    const startsWithMatch = RegExp('^' + regexp.source).exec(value)
+    if (startsWithMatch) return normalize(startsWithMatch)
+    const match = regexp.exec(value)
+    if (match) return normalize(match)
+  }
```

Comment:
Split into two loops: we prefer _any_ match at the beginning of the string to _any_ match later in the string.